tests/multisection_integration_tests/test_multi_sym_2_sec_constr.py

```diff
@@ -0,0 +1,147 @@
+        section_surfaces = build_sections(surface)
+        uniMesh = unify_mesh(section_surfaces)
+        surface["mesh"] = uniMesh
+
+        # Build a component with B-spline control points that joins the sections by construction
+        chord_comp = build_multi_spline("chord_cp", len(section_surfaces), sec_chord_cp)
+        prob.model.add_subsystem("chord_bspline", chord_comp)
+
+        # Connect the B-spline component to the section B-splines
+        connect_multi_spline(prob, section_surfaces, sec_chord_cp, "chord_cp", "chord_bspline")
+
+        # Create and add a group that handles the geometry for the
+        # aerodynamic lifting surface
+        multi_geom_group = MultiSecGeometry(surface=surface)
+        prob.model.add_subsystem(surface["name"], multi_geom_group)
+
+        # Create the aero point group, which contains the actual aerodynamic
+        # analyses
+        aero_group = AeroPoint(surfaces=[surface])
+        point_name = "aero_point_0"
+        prob.model.add_subsystem(
+            point_name, aero_group, promotes_inputs=["v", "alpha", "Mach_number", "re", "rho", "cg"]
+        )
+
+        # Get name of surface and construct unified mesh name
+        name = surface["name"]
+        unification_name = "{}_unification".format(surface["name"])
+
+        # Connect the mesh from the mesh unification component to the analysis point
+        prob.model.connect(
+            name + "." + unification_name + "." + name + "_uni_mesh", point_name + "." + "surface" + ".def_mesh"
+        )
+
+        # Perform the connections with the modified names within the
+        # 'aero_states' group.
+        prob.model.connect(
+            name + "." + unification_name + "." + name + "_uni_mesh",
+            point_name + ".aero_states." + "surface" + "_def_mesh",
+        )
+
+        # Add DVs
+        prob.model.add_design_var("chord_bspline.chord_cp_spline", lower=0.1, upper=10.0, units=None)
+        prob.model.add_design_var("alpha", lower=0.0, upper=10.0, units="deg")
+
+        # Add CL constraint
+        prob.model.add_constraint(point_name + ".CL", equals=0.3)
+        # prob.model.add_constraint(point_name +'.CL',upper=0.3,lower=0.3)
+
+        # Add Area constraint
+        prob.model.add_constraint(point_name + ".total_perf.S_ref_total", equals=2.0)
+
+        # Add objective
+        prob.model.add_objective(point_name + ".CD", scaler=1e4)
+
+        prob.driver = om.ScipyOptimizeDriver()
+        prob.driver.options["optimizer"] = "SLSQP"
+        prob.driver.options["tol"] = 1e-3
+        prob.driver.options["disp"] = True
+        prob.driver.options["maxiter"] = 1000
+        prob.driver.options["debug_print"] = ["nl_cons", "objs", "desvars"]
+
+        # Set up and run the optimization problem
+        prob.setup()
+        # prob.run_model()
+        prob.run_driver()
+        # om.n2(prob)
+
+        assert_near_equal(prob["aero_point_0.surface_perf.CD"][0], 0.02919978, 1e-6)
```

Comment:
Should the reference value be identical to the one in `test_multi_sym_2_sec.py` as these two look like running the same case with a different way of handling multi-section continuity? Or are they different in setup?

Reply:
It should be. They are fundamentally the same problem with a different way of handling the constraint. However, I think SLSQP has trouble with this constraint causing the optimum to differ by a small error 10^-5, not within the 10^-6 tolerance. SNOPT doesn't have this issue. I can either reduce the tolerance or just use different reference value.

Reply:
I think you should unify these two tests (test_multi_sym_2_sec.py and test_multi_sym_2_sec_constr.py) into one file/one class that runs two tests (i.e. has two methods). By doing so, you can reduce code duplication and set the reference value in one place (as an instance variable), which will be referred by the two tests. That way, you can test an output against the reference value and against the other way of handling geometry continuity.

And if SLSQP is causing the inconsistency here, I think you should use SNOPT for this test instead.

Reply:
It looks like you're setting different chord_cp values for these two tests. Unless it is intentional and needed, it'd be better to set the same values for both and make sure that both method gives identical outputs.

Reply:
You're right about the different intial chord_cp values. I made them and the same but it still gives the same result. I think it's an optimizer issue since SLSQP does not handle this particular constraint well.

Reply:
Issue is taken care of.